kroxylicious-operator/src/test/java/io/kroxylicious/operator/ProxyReconcilerTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.operator;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension;
+
+import io.kroxylicious.crapi.v1alpha1.KafkaProxy;
+import io.kroxylicious.crapi.v1alpha1.KafkaProxySpec;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+class ProxyReconcilerTest {
+
+    public static final String RESOURCE_NAME = "test-proxy";
+    public static final String INITIAL_BOOTSTRAP = "my-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092";
+    public static final String CHANGED_BOOTSTRAP = "your-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092";
+
+    @RegisterExtension
+    static LocallyRunOperatorExtension extension = LocallyRunOperatorExtension.builder()
+            .withReconciler(ProxyReconciler.class)
+            .build();
+
+    @Test
+    void testCRUDOperations() {
```

Comment:
Can we please split this into one test per operation at least!?

Reply:
I _guess_. The single test method alternative takes ~5-8s to execute. Believe it or not the three method version takes ~35s (dunno where the extra 10s comes from). So there's an incentive to not have too many test methods in this class. But it's liveable for now.